tests/queries/0_stateless/02503_test_cache_with_ttl_table.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest, no-s3-storage
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+ttl_on_columns="
+CREATE TABLE ttl (d DateTime, a String ttl d, b String ttl d)
+ENGINE = MergeTree
+ORDER BY tuple()
+PARTITION BY toSecond(d)
+SETTINGS min_bytes_for_wide_part = 0, storage_policy = 's3_cache', merge_with_ttl_timeout = 0;
+"
+ttl_on_table="
+CREATE TABLE ttl (d DateTime, a String, b String)
+ENGINE = MergeTree ORDER BY toDate(d) PARTITION BY tuple()
+TTL d
+SETTINGS remove_empty_parts = 1, storage_policy='s3_cache', merge_with_ttl_timeout = 0, min_bytes_for_wide_part = 0;
+"
+
+function insert()
+{
+${CLICKHOUSE_CLIENT} --multiline --multiquery -q """
+INSERT INTO ttl
+    SELECT now(), * FROM
+        generateRandom('a String, b String')
+    LIMIT 10000
+SETTINGS enable_filesystem_cache_on_write_operations = 0;
+"
+}
+
+function query()
+{
+${CLICKHOUSE_CLIENT} -q "SELECT * FROM ttl FORMAT Null;"
+}
+
+
+for create_table in "$ttl_on_columns" "$ttl_on_table"; do
```

Comment:
just a note: this test does not test the code changes above in cached read buffer, e.g. it works with/without those changes.